javascript/computer_science/project_hash_map.md

```diff
@@ -0,0 +1,223 @@
+### Introduction
+
+One of the most used data structure across all languages is a Hash Table, aka Hash Map. If you worked before with JavaScript Object Literal `{}`, `Set`, and `Map` then you are already familiar with a hash table. But how do they work exactly, how can we save values using strings and symbols as keys and then retrieving those values using the same keys.
+
+In this project, you will learn how it all works, and you will implement your own Hash Map! To start we need to learn a few concepts, one of which is a hash code, what does it mean, and how to use it. Let us get started and hope you are exited to finally know what is happening with an object when you put keys and values in them.
```

Comment:
There are a bunch of different ways "hash map" is written in this lesson. It appears as "hash map", "Hash Map", "HashMap", etc.

I think it'd be good to have a consistent approach. My preference would probably be "hash map" and then if we're referring to the class the user is making in the project, maybe `HashMap` can be fine too.